src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -0,0 +1,128 @@
+interface IParameters {
+  [key: string]: any;
```

Comment:
I had trouble when I tried to restrict this value's type. It can be any of `string`, `number` or `boolean`. And even when I defined it as being allowed to be any one of these, I was getting errors further down with the functions I tried to call on them if the method didn't work on ALL of these types (i.e. functions that only operate on strings gave an error saying it can't run on `boolean`, etc.)

What's the best practice for this kind of thing? Do a `if typeOf` check before calling each method and throw a custom error if not a match? Will typescript allow this?

Reply:
And also, if I try and restrict the type here, I also get an error on the `connection` field, as the `Client` type is not one of the allowed types for the `[key: string]` field(s). I can see the logic of why it's doing this, but not sure the best way to deal with it.